src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/FileRotate.java

```diff
@@ -115,7 +116,14 @@ protected Path rotate(long currentMillis) throws IOException {
       Files.move(FILE_TO_ROTATE, targetFilePath);
       lastRotatedMillis = System.currentTimeMillis();
     } catch (FileAlreadyExistsException fae) {
-      LOG.error(fae);
+      if (!Files.deleteIfExists(targetFilePath)) {
+        LOG.error("Could not delete file: " + targetFilePath);
+      }
+      try {
+        Files.move(FILE_TO_ROTATE, targetFilePath, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
We should add StandardCopyOption.ATOMIC_MOVE to the copy options as well.

Reply:
👍